### PR TITLE
globalDispatchFirst annotation

### PR DESCRIPTION
**Pull request by enumag**
This is a PR for #65.

### CODE REVIEW

File: src/Kdyby/Events/DI/EventsExtension.php [hunk 1]
```diff
@@ -42,11 +42,17 @@ class EventsExtension extends Nette\DI\CompilerExtension
 		'optimize' => TRUE,
 		'debugger' => '%debugMode%',
 		'exceptionHandler' => NULL,
+		'globalDispatchFirst' => FALSE,
 	);
 
 	/**
 	 * @var array
 	 */
+	private $config;
```
Comment (fprochazka) [inline src/Kdyby/Events/DI/EventsExtension.php]: ?

**Comment by fprochazka** [comment]
:+1:

**Comment by enumag** [comment]
@fprochazka I should probably add this to docs as well, shouldn't I?

**Comment by enumag** [comment]
@fprochazka By the way I just realized that this won't really help me. I'll have to refactor my code otherwise to do what I need in one transaction. That means I no longer have a use-case for this.

**Comment by fprochazka** [comment]
Doesn't matter, It's needed anyway and I'm going to review it and finish if neccesary :)

**Comment by michalfibek** [comment]
What's the status about this pull request? I stumbled upon the same event dispatch order problem today and I'd love to have this feature in master.

**Comment by hantc** [comment]
@michalfibek me too! :)

**Comment by hantc** [comment]
Btw, is there any way how to get these forked projects by composer? (and of course with specific branch 'listeners')

**Comment by enumag** [comment]
@hantc Sure, use this in composer.json:

```
	"require": {
		"kdyby/events": "dev-listeners as 2.4.0",
	},
	"repositories": [
		{
			"type": "vcs",
			"url": "https://github.com/enumag/Events.git"
		}
	],
```

**Comment by enumag** [comment]
@fprochazka The first two commits should be merged anyway. Should I prepare separate PR?

**Comment by fprochazka** [comment]
@enumag I've cherrypicked them. Thank you. I have lost track of this PR, if you wanna, it would be helpful if you could finish it, otherwise I'll do it "sometime" :)

**Comment by enumag** [comment]
@fprochazka If there is something unfinished in this PR I'm not aware of it. Personally I'd close this without merging but if you still wish to add it I'm willing to help.

**Comment by fprochazka** [comment]
Thanks, merged https://github.com/Kdyby/Events/commit/fc82cd11e3415bb3c30d661948ab6279c40caa04

**Pull request closed without merging**